### PR TITLE
Add Gemma3 to MaxText

### PR DESCRIPTION
**Pull request by gagika**
# Description

Add Gemma3 model in MaxText (Text-only) version

# Tests

Tests are included e2e tests with logits check for Gemma3-4b, Gemma3-12b and Gemma3-27 models.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by khatwanimohit — APPROVED** [review]
Go Gemma!! 🚀 ❤️ 💎

**Review by shralex — APPROVED** [review]

**Review by gobbleturk — APPROVED** [review]

**Review by bvandermoon — APPROVED** [review]
Awesome!! :fire:
